Framework/PythonInterface/test/python/mantid/kernel/ArrayPropertyTest.py

```diff
@@ -129,10 +129,44 @@ def PyInit(self):
                     FloatArrayProperty("Input", Direction.Input), "Float array")
             def PyExec(self):
                 self._input_values = self.getProperty("Input").value
-    
+
         input_values = range(1, 5)
         self._do_algorithm_test(AlgWithFloatArrayProperty, input_values)
 
+    def test_dtype_function_calls(self):
+        """
+        Tests the dtype() function call for the data types stored in the array.
+        """
+        # Set up
+        direc = Direction.Output
+        validator = NullValidator()
+
+        # Create float array
+        float_input_values = [1.1, 2.5, 5.6, 4.6, 9.0, 6.0]
+        float_arr = FloatArrayProperty("floats", float_input_values, validator, direc)
+
+        # Create int array
+        int_input_values = [1, 2, 5, 4, 9, 6]
+        int_arr = IntArrayProperty("integers", int_input_values, validator, direc)
+
+        # Create string array
+        str_input_values =["a", "b", "c", "d", "e"]
+        str_arr = StringArrayProperty("letters", str_input_values, validator, direc)
+
+        # Test
+        self.assertEquals(float_arr.dtype(), "double")
+        self.assertEquals(str_arr.dtype(), "string")
+        
+        # Implementation of IntArrayProperty is based on long 
+        # (which is itself implementation defined, so a special case is needed here)
+        if sys.platform == 'win32' or sys.platform == 'darwin':
+          # Windows and macOS should return "long"
+          self.assertEquals(int_arr.dtype(), "long")
+        else:
+          # Linux based systems should return "int64"
+          self.assertEquals(int_arr.dtype(), "int64")
+
+
```

Comment:
This is good, but I would suggest another test be added that constructs a numpy array with the given dtype. This checks that they strings we return are compliant with what numpy expects.